src/main/kotlin/org/rust/lang/core/psi/RustLiteral.kt

```diff
@@ -1,14 +1,88 @@
 package org.rust.lang.core.psi
 
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElementVisitor
 import com.intellij.psi.impl.source.tree.LeafPsiElement
 import com.intellij.psi.tree.IElementType
+import org.rust.lang.core.psi.visitors.RustVisitorEx
 
 sealed class RustLiteral(type: IElementType, text: CharSequence) : LeafPsiElement(type, text) {
-    abstract class Number(type: IElementType, text: CharSequence) : RustLiteral(type, text)
-    abstract class Text(type: IElementType, text: CharSequence) : RustLiteral(type, text)
+    /**
+     * Get literal value as a Java object.
+     */
+    abstract val value: Any?
```

Comment:
So I have mixed feelings about this. Making `RustLiteral` generic over literal's value type counterpart in Java will not only introduce additional verbosity (`RustLiteral<*>` instead of `RustLiteral`), but also doesn't add any special meaning to the type itself. `value` is just an utility, not integral part of the literal.